app/services/fhir/references/reference_misc.py

```diff
@@ -1,19 +1,28 @@
+
 from fhir.resources.R4B.reference import Reference
+import logging
+
+from app.models.adjacency.node import NodeReference
 
+logger = logging.getLogger(__name__)
 
-def split_reference(data: Reference) -> tuple[str, str]:
+def build_node_reference(data: Reference, base_url: str) -> NodeReference:
     """
-    Returns a split resource from a reference (e.g. "Patient/123" -> ("Patient", "123"))
+    Converts a FHIR Reference to a NodeReference, making it relative to the given base URL if necessary.
     """
     ref = data.reference
     if ref is None:
-        raise ValueError("Cannot parse an None Value")
+        raise ValueError("Invalid reference (None)")
 
-    parts = ref.split("/")
-    if len(parts) < 2:
-        raise ValueError(
-            "Invalid Reference type, this should be a valid relative or absolute URL"
-        )
+    if ref.startswith(base_url):
+        ref = ref[len(base_url):].lstrip("/")
 
-    return parts[-2], parts[-1]
+    if ref.startswith("https://") or ref.startswith("http://"):
```

Comment:
Add the Sonarqube ignore thingy
```suggestion
    if ref.startswith("https://") or ref.startswith("http://"): # NOSONAR
```